org.opent2t.sample.hub.superpopular/com.insteon.hub/js/thingTranslator.js

```diff
@@ -0,0 +1,300 @@
+            var promises = [];
+            devices.forEach((insteonDevice) => {
+                
+                var promise = this._makeRequest(this._devicesPath + '/' + insteonDevice.DeviceID, 'GET')
+                    .then((details) => {
+
+                        // get the opent2t schema and translator for the insteon device
+                        var opent2tInfo = this._getOpent2tInfo(details.DevCat, details.SubCat.toString(16).toUpperCase());
+
+                        if (opent2tInfo != undefined) // we support the device
+                        {
+                            var device = {};
+
+                            // we only need to return certain properties back
+                            device.name = details.DeviceName;
+
+                            // set the specific device object id to be the id
+                            device.id = details.DeviceID;
+
+                            // set the opent2t info for the insteon device
+                            device.openT2T = opent2tInfo;
+                            
+                            return Promise.resolve(device);
+                        }
+                        return Promise.resolve(undefined);
+                    });
+
+                promises.push(promise);
+            });
+
+            return Promise.all(promises).then((results) => {
+                for (var i = 0; i < results.length ; i++) {
+                    if(results[i] !== undefined){
+                        filteredDevices.push(results[i]);
+                    }
+                }
+
+                toReturn.devices = filteredDevices;
+                return Promise.resolve(toReturn);
+            });
+        });
+    }
+
+    /**
+     * Get the name of the hub.  Ties to the n property from oic.core
+     */
+    getN() {
+        return this._name;
+    }
+
+    /**
+     * Gets device details (all fields), response formatted per http://docs.insteon.apiary.io/#reference/devices
+     */
+    getDeviceDetailsAsync(deviceId) {
+
+        // Make the async request
+        return this._makeRequest(this._devicesPath + '/' + deviceId, 'GET')
+            .then((details) => {
+
+                var device = details;
+
+                var postPaylaod =
+                {
+                    command: 'get_status',
+                    device_id: deviceId
+                };
+
+                return this._makeRequest(this._commandPath, 'POST', JSON.stringify(postPaylaod))
+                    .then((response) => {                       
+                        return this._getCommandResponse(response.id, 3) // get device status
+                            .then((deviceStatus) => {
+
+                                if (deviceStatus !== undefined && deviceStatus.status == 'succeeded') {   
+                                    device['Level'] = deviceStatus.response.level;
+                                    if (deviceStatus.response.level == 0) {
```

Comment:
it may not matter in this case, javascript best practice is to use === to be safe

---
In reply to: [84631900](https://github.com/openT2T/translators/pull/107#discussion_r84631900) [](ancestors = 84631900,82905202)